specification/flag-evaluation/flag-evaluation.md

```diff
@@ -0,0 +1,195 @@
+});
+```
+
+The name is a logical identifier for the client.
+
+##### Requirement 1.6
+
+> The client **MUST** provide a method for `hook` addition, which accepts an array of API-conformant `hooks`. This adds the specified hooks to any hooks already attached.
+
+```
+// example hook attachment
+client.addHooks([new MyHook()]);
+```
+
+see [hooks](./hooks.md)
+
+#### Flag Evaluation
+
+##### Requirement 1.7
+
+> The `client` **MUST** provide methods for typed flag value evaluation, including boolean, numeric, string, and map (object), with parameters `flag key` (string, required), `default value` (boolean | number | string | object, required), `evaluation context` (optional), and `evaluation options` (optional).
+
+```
+// example boolean flag evaluation
+boolean myBool =  client.getBooleanValue('bool-flag', false);
+
+
+// example overloaded string flag evaluation with optional params
+string myString = client.getStringValue('string-flag', 'N/A', evaluationContext, options);
+
+```
+
+see [evaluation context](../evaluation-context/evaluation-context.md)
+
+##### Requirement 1.8
+
+> The `client` **MUST** provide a method for flag activity evaluation, denoting if the flag is "enabled" or not, with parameters `flag key` (string, required), `default value` (boolean, required), `evaluation context` (optional), and `evaluation options` (optional).
+
+```
+// example flag activity evaluation
+boolean enabled = client.isEnabled('my-flag', false);
+```
+
+##### Requirement 1.9
+
+> The `client` method for flag activity evaluation **MAY** be semantically different from the boolean flag evaluation method.
+
+Some providers may draw a distinction between boolean flag value evaluation, and flag activity. Put another way, a flag may be "disabled", but simultaneously have an associated auxillary boolean payload of `true`.
+
+##### Requirement 1.10
+
+> The `client` **MUST** provide methods for flag value evaluation with the value wrapped in a `evaluation details` object, with parameters `flag key` (string, required), `default value` (boolean | number | string | object, required), `evaluation context` (optional), and `evaluation options` (optional).
+
+```
+// example detailed boolean flag evaluation
+FlagEvaluationDetails<boolean> myBoolDetails = client.getBooleanDetails('bool-flag', false);
+
+// example detailed string flag evaluation
+FlagEvaluationDetails<string> myStringDetails = client.getStringDetails('string-flag', 'N/A', evaluationContext, options);
+
+```
+
+##### Requirement 1.11
+
+> The `evaluation details` wrapper **MUST** have a `value` field matching the type of the flag value, and containing to the flag value.
+
+##### Condition 1.12
+
+> The language supports generics (or an equivalent feature).
+>
+> ##### Conditional Requirement 1.12.1
+>
+> > The `evaluation details` wrapper **SHOULD** accept a generic argument (or utilize an equivalent language feature) which indicates the type of the wrapped `value` field.
```

Comment:
What happens if there is a type mismatch?

Reply:
I was going to mention more about this in this provider spec, but I should stay in this spec that all the flag evaluation methods should default when there's a type mismatch (for the provider spec, the provider should throw or otherwise indicate abnormal execution).